httpx/_utils.py

```diff
@@ -134,9 +134,12 @@ def netrc_info(self) -> typing.Optional[netrc.netrc]:
             self._netrc_info = None
             for file_path in self.netrc_files:
                 expanded_path = Path(file_path).expanduser()
-                if expanded_path.is_file():
-                    self._netrc_info = netrc.netrc(str(expanded_path))
-                    break
+                try:
+                    if expanded_path.is_file():
+                        self._netrc_info = netrc.netrc(str(expanded_path))
```

Comment:
We should probably place the try/except block more precisely, ie around this particular line? (I assume the `.is_file` check isn’t what’s causing the `PermissionError`.)

Reply:
I'd assume `is_file` probably *could* raise an IOError here if there's no permissions to that path?

Reply:
Hmm, yes, that’s something we’d need to double check — is that covered by `PermissionError`?

Reply:
Here is the trace info:

[2020-02-11 22:29:24,797] ERROR in app: Exception on /my/app/url/path [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.7/dist-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.7/dist-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/dist-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.7/dist-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.7/dist-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/path/to/my/app.py", line 28, in my_function
    my_variable=my_variable.my_function(my_variable,my_variable)
  File "/path/to/my/app.py", line 17, in my_function
    my_variable=my_variable.my_function()
  File "/path/to/my/app.py", line 73, in my_function
    r=httpx.get(url,cookies=self.cookies,headers=headers)
  File "/usr/local/lib/python3.7/dist-packages/httpx/api.py", line 170, in get
    trust_env=trust_env,
  File "/usr/local/lib/python3.7/dist-packages/httpx/api.py", line 96, in request
    allow_redirects=allow_redirects,
  File "/usr/local/lib/python3.7/dist-packages/httpx/client.py", line 573, in request
    request, auth=auth, allow_redirects=allow_redirects, timeout=timeout,
  File "/usr/local/lib/python3.7/dist-packages/httpx/client.py", line 590, in send
    auth = self.build_auth(request, auth)
  File "/usr/local/lib/python3.7/dist-packages/httpx/client.py", line 282, in build_auth
    credentials = self.netrc.get_credentials(request.url.authority)
  File "/usr/local/lib/python3.7/dist-packages/httpx/utils.py", line 145, in get_credentials
    if self.netrc_info is None:
  File "/usr/local/lib/python3.7/dist-packages/httpx/utils.py", line 137, in netrc_info
    if expanded_path.is_file():
  File "/usr/lib/python3.7/pathlib.py", line 1365, in is_file
    return S_ISREG(self.stat().st_mode)
  File "/usr/lib/python3.7/pathlib.py", line 1161, in stat
    return self._accessor.stat(self)
PermissionError: [Errno 13] Permission denied: '/root/.netrc'